### ISSUE TITLE
Build hangs on "creating build/temp.linux-x86_64-3.6/src"

### ISSUE THREAD

**Issue by gridley** [mit-crpg/OpenMOC#390]
Hi all,

I'm trying to build OpenMOC on the develop branch. The relevant output is below. It's on ubuntu 18.04.

```
gavin@gdesk:~/code/OpenMOC$ python3 setup.py clean
running clean
removing 'build/temp.linux-x86_64-3.6' (and everything under it)
gavin@gdesk:~/code/OpenMOC$ python3 setup.py install --verbose --user --with-cuda
running install
running build
running build_py
copying openmoc/openmoc.py -> build/lib/openmoc
copying openmoc/cuda/openmoc_cuda.py -> build/lib/openmoc/cuda
running build_ext
/usr/share/swig3.0/python/pyiterators.swg:401: Warning 511: Can't use keyword arguments with overloaded functions (swig::SwigPyIterator::operator -(ptrdiff_t) const).
/usr/share/swig3.0/python/pyiterators.swg:402: Warning 511: Can't use keyword arguments with overloaded functions (swig::SwigPyIterator::operator -(swig::SwigPyIterator const &) const).
building '_openmoc' extension
creating build/temp.linux-x86_64-3.6
creating build/temp.linux-x86_64-3.6/openmoc
creating build/temp.linux-x86_64-3.6/src
```
How can this be diagnosed?

**Comment by GiudGiud** [comment]
Hi Gavin

I think @wuwenbin2006 ran into a similar issue before. @wuwenbin2006 was it the same problem ? Did you find a solution ?

@gridley what are the versions of all the dependencies (swig, h5py, distutils) and compiler ? Are you working with a VM ?
Try strace (-p) setup.py install --verbose --user --with-cuda
It will provide a lot of output, but you should be able to see what's not returning.

Best,
Guillaume

**Comment by gridley** [comment]
Thanks for the response Guillaume. Here's the setup info.

```
gavin@gdesk:~/code/OpenMOC$ swig -version

SWIG Version 3.0.12

Compiled with g++ [x86_64-pc-linux-gnu]

Configured options: +pcre

Please see http://www.swig.org for reporting bugs and further information
gavin@gdesk:~/code/OpenMOC$ python3 -c 'import h5py, distutils; print(h5py.__version__); print(distutils.__version__)'
2.9.0
3.6.7
gavin@gdesk:~/code/OpenMOC$ g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.3.0-27ubuntu1~18.04' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.3.0 (Ubuntu 7.3.0-27ubuntu1~18.04) 
gavin@gdesk:~/code/OpenMOC$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Fri_Nov__3_21:07:56_CDT_2017
Cuda compilation tools, release 9.1, V9.1.85
```

As for strace, the last three calls are:

```
futex(0x7fb904001aa0, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0xa73588, FUTEX_WAKE_PRIVATE, 1)  = 1
futex(0xa73520, FUTEX_WAKE_PRIVATE, 1)  = 1
futex(0xc3b820, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, NULL, 0xffffffff
```

Not sure why this futex thing would hold stuff up though.

**Comment by GiudGiud** [comment]
Ok I'll see if I can reproduce the bug by updating to your versions. 
Does it compile without the --with-cuda flag ?
strace was too fine grained I guess.
Try `gdb -batch -ex "run" -ex "bt" python setup.py install --verbose --user --with-cuda`

**Comment by gridley** [comment]
I'm away from the computer having this issue now, but I can confirm going without cuda yields the same issue. Thanks for looking into this very much!

**Comment by wuwenbin2006** [comment]
Hi, Gavin and Guil Yes, I was stuck in the build process once. I just went back from ubuntu 18 to ubuntu 16, and the default python3 version changed from 3.7 or 3.6 down to 3.5. Then it worked. I was building it on Wndows 10  subsystem Linux. Hope this helps. Wenbin Wu On 02/20/2019 02:13, Gavin Ridley wrote: I'm away from the computer having this issue now, but I can confirm going without cuda yields the same issue. Thanks for looking into this very much! — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by GiudGiud** [comment]
@wuwenbin2006 thanks for letting us know
@gridley Upgrading gcc, swig, distutils to your versions didn't reproduce the problem for me. I'll try to find a machine with Ubuntu 18.04, will let you know

**Comment by GiudGiud** [comment]
Fixed by #391 . Fix will be included in 3D-MOC by a future PR
